main.c

```diff
@@ -0,0 +1,32 @@
+#include "ocollection.h"
```

Comment:
your header file isn't added to your build system (see https://cmake.org/cmake/help/v3.16/guide/tutorial/index.html#a-basic-starting-point-step-1)